nixos/modules/tasks/filesystems/bcachefs.nix

```diff
@@ -1,26 +1,52 @@
-{ config, lib, pkgs, ... }:
+{ config, lib, pkgs, utils, ... }:
 
 with lib;
 
 let
 
-  inInitrd = any (fs: fs == "bcachefs") config.boot.initrd.supportedFilesystems;
+  bootFs = filterAttrs (n: fs: (fs.fsType == "bcachefs") && (utils.fsNeededForBoot fs)) config.fileSystems;
 
-in
-
-{
-  config = mkIf (any (fs: fs == "bcachefs") config.boot.supportedFilesystems) {
-
-    system.fsPackages = [ pkgs.bcachefs-tools ];
-
-    # use kernel package with bcachefs support until it's in mainline
-    boot.kernelPackages = pkgs.linuxPackages_testing_bcachefs;
-    boot.initrd.availableKernelModules = mkIf inInitrd [ "bcachefs" ];
-
-    boot.initrd.extraUtilsCommands = mkIf inInitrd
+  openCommand = name: fs: 
+    let
+      firstDevice = head (splitString ":" fs.device); # we need only unlock one device manually, and cannot pass multiple at once
```

Comment:
This still seems very weird to me, since you can't pass in by-path devices, but I assume this is a limitation of upstream?

Reply:
You're absolutely right to mention this - it was rather late yesterday when I got the information and removed the stage 1 patch. I have now changed it so we try to unlock once with splitting and once without (until upstream adds filesystem UUID mounting).

Reply:
Yes, this is a limitation of upstream, as per the comment below.